src/makeInfinityAdder.js

```diff
@@ -5,6 +5,23 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let sum = 0;
+
+  function adder(arg) {
+    if (arg === undefined) {
```

Comment:
The check for `undefined` is correct for determining when to return the result. However, consider adding a type check to ensure that the argument, if provided, is a number to avoid unexpected behavior.